### ISSUE TITLE
False positive warning on the use of size aesthetics on geom_sf()

### ISSUE THREAD

**Issue by yutannihilation** [tidyverse/ggplot2#4960]
I think this is minor (and not sure if this can be fixed), so it should be fine if this won't be fixed before the release.

``` r
library(ggplot2)

pts <- sf::st_sf(
  geometry = sf::st_sfc(sf::st_point(0:1), sf::st_point(1:2)),
  size = c(1, NA),
  shape = c("a", NA),
  colour = c("red", NA)
)

# OK
b1 <- ggplot_build(ggplot(pts) + geom_sf(aes(size = size)))

# False positive
b2 <- ggplot_build(ggplot(pts, aes(size = size)) + geom_sf())
#> size aesthetic has been deprecated for use with lines as of ggplot2 3.4.0
#> ℹ Please use linewidth aesthetic instead
#> This message is displayed once every 8 hours.
```

<sup>Created on 2022-08-24 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by yutannihilation** [comment]
I noticed this because this line shows the warning.

https://github.com/tidyverse/ggplot2/blob/c86ed7bc59060d94fb070fe1bb1574e8b6026ff8/tests/testthat/test-geom-sf.R#L104

**Comment by yutannihilation** [comment]
Resolved by #4964
